internal/repository/master_index.go

```diff
@@ -237,6 +242,31 @@ func (mi *MasterIndex) Each(ctx context.Context) <-chan restic.PackedBlob {
 	return ch
 }
 
+// MergeFinalIndexes merges all final indexes together.
+// After calling, there will be only one big final index in MasterIndex
+// containing all final index contents.
+// Indexes that are not final are left untouched.
+func (mi *MasterIndex) MergeFinalIndexes() {
```

Comment:
I would be much more convenient if a new index is merged automatically when a final index is added via `Insert`.

Reply:
Yes, would be. There is however a problem if taking care of superseded index files: If treated correctly, they should be removed from MasterIndex. However when reading all index files, one already superseded index could be read and inserted while the index which supersedes that index will be read and inserted later. So in this case we would have to remove the entries again. This is why I first read and Insert all index files (and `Insert` can easily take care of deleting "newly" superseded ones) and then merge the final indexes.

A correct implementation of treating superseded indexes in `Insert` is in #2718 - however I think, I should make a separate PR out of it.